### PR TITLE
nixos/grub: make memtest work with EFI

### PR DESCRIPTION
**Pull request by wedens**
Memtest86+ doesn't support EFI, so unfree Memtest86 is used when EFI
support is enabled (systemd-boot currently also uses Memtest86 when
memtest is enabled).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by worldofpeace** [comment]
Wouldn't this make every person with EFI grub have to do `allowUnfree` all of a sudden?

File: nixos/modules/system/boot/loader/grub/memtest.nix [hunk 1]
```diff
@@ -75,19 +79,38 @@ in
...
-        ''
-      else
-        throw "Memtest86+ is not supported with GRUB 1.";
-
-    boot.loader.grub.extraFiles."memtest.bin" = "${memtest86}/memtest.bin";
+  config = mkMerge [
+    (mkIf (cfg.enable && efiSupport) {
+      assertions = [
+        {
+          assertion = cfg.params != [];
```
Comment (Frostman) [inline nixos/modules/system/boot/loader/grub/memtest.nix]: It should be inverted - assertion should be true for the check to pass.

**Review by Frostman — CHANGES_REQUESTED** [review]

File: nixos/modules/system/boot/loader/grub/memtest.nix (hunk 1 above)
Comment (wedens) [reply nixos/modules/system/boot/loader/grub/memtest.nix]: Fixed

**Review by samueldr — APPROVED** [review]
Tested using:

 * https://gist.github.com/samueldr/aed4193cd4deaa1d8de91a2ad18c2c26

**Comment by worldofpeace** [comment]
Hmm, does this need a release note...

**Comment by samueldr** [comment]
It could, as a nice new thing, but doesn't require one, imo.

Given that it's unfree, maybe not.

**Pull request merged**